services/skus/controllers_noint_test.go

```diff
@@ -222,3 +223,151 @@ func TestHandleReceiptErr(t *testing.T) {
 		})
 	}
 }
+
+func TestParseVerifyCredRequestV2(t *testing.T) {
+	type tcExpected struct {
+		val   *model.VerifyCredentialRequestV2
+		errFn must.ErrorAssertionFunc
+	}
+
+	type testCase struct {
+		name  string
+		given []byte
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name:  "error_malformed_payload",
+			given: []byte(`nonsense`),
+			exp: tcExpected{
+				errFn: func(tt must.TestingT, err error, i ...interface{}) {
+					must.Equal(tt, true, err != nil)
+				},
+			},
+		},
+
+		{
+			name:  "error_malformed_credential",
+			given: []byte(`{"sku":"sku","merchantId":"merchantId"}`),
+			exp: tcExpected{
+				errFn: func(tt must.TestingT, err error, i ...interface{}) {
+					must.Equal(tt, true, err != nil)
+				},
+			},
+		},
+
+		{
+			name:  "success_complete",
+			given: []byte(`{"sku": "sku","merchantId": "merchantId","credential":"eyJ0eXBlIjoidGltZS1saW1pdGVkLXYyIiwicHJlc2VudGF0aW9uIjoiVG1GMGRYSmxJR0ZpYUc5eWN5QmhJSFpoWTNWMWJTNEsifQo="}`),
+			exp: tcExpected{
+				val: &model.VerifyCredentialRequestV2{
+					SKU:        "sku",
+					MerchantID: "merchantId",
+					Credential: "eyJ0eXBlIjoidGltZS1saW1pdGVkLXYyIiwicHJlc2VudGF0aW9uIjoiVG1GMGRYSmxJR0ZpYUc5eWN5QmhJSFpoWTNWMWJTNEsifQo=",
+					CredentialOpaque: &model.VerifyCredentialOpaque{
+						Type:         "time-limited-v2",
+						Presentation: "TmF0dXJlIGFiaG9ycyBhIHZhY3V1bS4K",
+					},
+				},
+				errFn: func(tt must.TestingT, err error, i ...interface{}) {
+					must.Equal(tt, true, err == nil)
+				},
+			},
+		},
+	}
+
+	for i := range tests {
+		tc := tests[i]
+
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := parseVerifyCredRequestV2(tc.given)
+			tc.exp.errFn(t, err)
```

Comment:
nit: naming the function `mustErr` opposed to `errFn` would give more indication that this will not continue and will perform an assertion.

Reply:
I thought about this, and liked `errFn` more, and chose it. However, updated according to your suggestion. Thanks.